### DISCUSSION TITLE
How to break a self-intesect polygon or a relatively simple polygon into difference polygon?

### DISCUSSION THREAD

**Discussion by jt-chairman** [CGAL/cgal#7850] [Q&A] [answered]
Here is a simple example I construct a relatively simple polygon.

` 

       Polygon_2 outer;

        outer.push_back(Point(0, 0));

        outer.push_back(Point(8, 0));

        outer.push_back(Point(8, 8));

        outer.push_back(Point(4, 0));

        outer.push_back(Point(0, 8));
`

Is there any API I can use to break this polygon into two triangle? The reason I what to do this is it will return a nullptr when I call _create_interior_straight_skeleton_2()_ to this kind of polygon.

**Comment by sloriot** [comment, marked as answer]
Maybe this new feature will help: https://github.com/CGAL/cgal/pull/7608

**Reply by jt-chairman** [reply]
WOW! new package! When does this new feature will be released?
Also, It there any trick that we can combine the currently available API to get through this?

**Reply by sloriot** [reply]
It needs to be reviewed and approved by the editorial board so I cannot tell you about the official release date.
For using the new code, you can checkout the branch and simply set `CGAL_DIR` in cmake to the checkout directory of the branch and you should be all set.

**Reply by jt-chairman** [reply]
Thanks for your reply!
